### PR TITLE
Sampling

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

**Comment by pavolloffay** [comment]
@objectiser could you please have at this?

File: docs/contributors/opentracingprovider.adoc [hunk 1]
```diff
@@ -188,4 +189,34 @@ When the ‘inject’ method is called on the OpenTracing Tracer, it will provid
 An unique id should be created, associated with the trace state being returned for injection into the outbound message, and used to create an `Interaction` based correlation id for the current node. The node should also be created as a `Producer`.
 
 
+=== Sampling 
 
+Sampling implementation is not required, however if a provider decideds to implement it should follow following rules.
```
Comment (objectiser) [inline docs/contributors/opentracingprovider.adoc]: Not sure if we should make it optional - wouldn't it be better to have it in all implementations for consistency?

Comment (objectiser) [reply docs/contributors/opentracingprovider.adoc]: This is my only question - if you think it should be optional, then "decideds' -> 'decides' and 'follow' change to 'adhere to the'.

**Review by objectiser — COMMENTED** [review]
Only the one minor comment.

File: docs/contributors/opentracingprovider.adoc (hunk 1 above)
Comment (pavolloffay) [reply docs/contributors/opentracingprovider.adoc]: I think that openTracing provider can be used without sampling, and probably users can provide their own sampling algorithms. 

Or maybe should we say that if the sampling is not implemented that provider should explicitly set  `ReportingLevel` to `All`?

Comment (objectiser) [reply docs/contributors/opentracingprovider.adoc]: We need to distinguish between whether an opentracing provider should support sampling vs whether the application needs to provide a sampling configuration. I think we should ensure all the APM OT providers support sampling using the same common API - however the application does not need to provide it, and if they don't provide, (and a reporting level is not propagated) then it should default to All.

Comment (pavolloffay) [reply docs/contributors/opentracingprovider.adoc]: This section talks about implementing opentracing provider right? 

What I wanted to stress is that is somebody implements a provider without sampling then the provider should set reporting level to 'All' so by default all providers should implement some kind of `AlwaysSampler` I will update the doc

Comment (objectiser) [reply docs/contributors/opentracingprovider.adoc]: ok thanks +1

**Comment by pavolloffay** [comment]
@objectiser PR updated could you please look?

**Comment by objectiser** [comment]
@pavolloffay LGTM :+1:  Thanks!

**Pull request merged**